src/goto-symex/symex_main.cpp

```diff
@@ -159,58 +161,87 @@ void goto_symext::symex_threaded_step(
   }
 }
 
-/// symex from given state
-void goto_symext::operator()(
+void goto_symext::symex_with_state(
   statet &state,
   const goto_functionst &goto_functions,
-  const goto_programt &goto_program)
+  const goto_programt &goto_program,
+  symbol_tablet &new_symbol_table)
 {
+  ns=namespacet(outer_symbol_table, state.symbol_table);
```

Comment:
As above, could you comment why namespaces are being re-initialised to point at different tables like this? What tables will `ns` refer to and at what stage in proceedings?